packages/graphql/src/api-v6/queryIR/DeleteOperation.ts

```diff
@@ -0,0 +1,189 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Cypher from "@neo4j/cypher-builder";
+import type { ConcreteEntityAdapter } from "../../schema-model/entity/model-adapters/ConcreteEntityAdapter";
+import type { InterfaceEntityAdapter } from "../../schema-model/entity/model-adapters/InterfaceEntityAdapter";
+import type { Filter } from "../../translate/queryAST/ast/filters/Filter";
+import type { DeleteOperation } from "../../translate/queryAST/ast/operations/DeleteOperation";
+import type { OperationTranspileResult } from "../../translate/queryAST/ast/operations/operations";
+import { MutationOperation } from "../../translate/queryAST/ast/operations/operations";
+import type { QueryASTContext } from "../../translate/queryAST/ast/QueryASTContext";
+import type { QueryASTNode } from "../../translate/queryAST/ast/QueryASTNode";
+import type { EntitySelection, SelectionClause } from "../../translate/queryAST/ast/selection/EntitySelection";
+import { wrapSubqueriesInCypherCalls } from "../../translate/queryAST/utils/wrap-subquery-in-calls";
+
+export class V6DeleteOperation extends MutationOperation {
+    public readonly target: ConcreteEntityAdapter | InterfaceEntityAdapter;
+    private selection: EntitySelection;
+    private filters: Filter[];
+    private nestedOperations: MutationOperation[];
+
+    constructor({
+        target,
+        selection,
+        nestedOperations = [],
+        filters = [],
+    }: {
+        target: ConcreteEntityAdapter | InterfaceEntityAdapter;
+        selection: EntitySelection;
+        filters?: Filter[];
+        nestedOperations?: DeleteOperation[];
+    }) {
+        super();
+        this.target = target;
+        this.selection = selection;
+        this.filters = filters;
+        this.nestedOperations = nestedOperations;
+    }
+
+    public getChildren(): QueryASTNode[] {
+        return [this.selection, ...this.filters, ...this.nestedOperations];
+    }
+
+    public transpile(context: QueryASTContext): OperationTranspileResult {
+        if (!context.hasTarget()) {
+            throw new Error("No parent node found!");
+        }
+        const { selection, nestedContext } = this.selection.apply(context);
+        if (nestedContext.relationship) {
+            return this.transpileNested(selection, nestedContext);
+        }
+        return this.transpileTopLevel(selection, nestedContext);
+    }
+
+    private transpileTopLevel(
+        selection: SelectionClause,
+        context: QueryASTContext<Cypher.Node>
+    ): OperationTranspileResult {
+        this.validateSelection(selection);
+        const filterSubqueries = wrapSubqueriesInCypherCalls(context, this.filters, [context.target]);
+        const predicate = this.getPredicate(context);
+        const extraSelections = this.getExtraSelections(context);
+
+        const nestedOperations: (Cypher.Call | Cypher.With)[] = this.getNestedDeleteSubQueries(context);
```

Comment:
nitpick/question. Isn't the type `Array<Cypher.Call | Cypher.With>` a little bit easier to read?

Reply:
Also, I think these nested operations are not tested, AFAIK the current scope of the top level delete does not need this code. The code looks fine but I'm a bit wary of adding code that is not run. Maybe add this code in the nested delete PR if that is something you are already working on?